File: mobile/app/src/main/java/com/praeter/data/remote/api/UserApiService.java
package com.praeter.data.remote.api;

import com.praeter.data.remote.dto.ApiResponse;
import com.praeter.data.remote.dto.User;

import io.reactivex.rxjava3.core.Single;
import retrofit2.http.Body;
import retrofit2.http.PATCH;
import retrofit2.http.POST;

public interface UserApiService {

    @POST("/users")
    Single<ApiResponse> saveUser(@Body User user);

    @PATCH("/users/login")
    Single<ApiResponse> login(@Body User user);
}

File: mobile/app/src/main/java/com/praeter/data/remote/api/UserApiService.kt
package com.praeter.data.remote.api

import com.praeter.data.remote.dto.ApiResponse
import com.praeter.data.remote.dto.user.User
import retrofit2.http.Body
import retrofit2.http.PATCH
import retrofit2.http.POST

interface UserApiService {
    @POST("/users")
    suspend fun saveUser(@Body user: User): ApiResponse

    @PATCH("/users/login")
    suspend fun login(@Body user: User): ApiResponse
}